be/src/exec/es/es_scan_reader.cpp

```diff
@@ -49,20 +51,37 @@ ESScanReader::ESScanReader(const std::string& target, const std::map<std::string
     if (props.find(KEY_QUERY) != props.end()) {
         _query = props.at(KEY_QUERY);
     }
+
     std::string batch_size_str = props.at(KEY_BATCH_SIZE);
     _batch_size = atoi(batch_size_str.c_str());
-    _init_scroll_url = _target + REQUEST_SEPARATOR + _index + REQUEST_SEPARATOR + _type + "/_search?scroll=" + _scroll_keep_alive + REQUEST_PREFERENCE_PREFIX + _shards + "&" + REUQEST_SCROLL_FILTER_PATH;
-    _next_scroll_url = _target + REQUEST_SEARCH_SCROLL_PATH + "?" + REUQEST_SCROLL_FILTER_PATH;
+
+    if (props.find(KEY_TERMINATE_AFTER) != props.end()) {
+        _exactly_once = true;
+        _terminate_after = props.at(KEY_TERMINATE_AFTER);
```

Comment:
seems that a local variable is enough for _terminate_after

Reply:
Yeah.